source/common/upstream/outlier_detection_impl.cc

```diff
@@ -660,6 +749,15 @@ void EventLoggerImpl::logEject(const HostDescriptionConstSharedPtr& host, Detect
         detector.successRateEjectionThreshold(monitor_type));
     event.mutable_eject_success_rate_event()->set_host_success_rate(
         host->outlierDetector().successRate(monitor_type));
+  } else if ((type == envoy::data::cluster::v2alpha::OutlierEjectionType::FAILURE_PERCENTAGE) ||
+             (type == envoy::data::cluster::v2alpha::OutlierEjectionType::
```

Comment:
Looks like this could use unit tests?

https://272990-65214191-gh.circle-artifacts.com/0/coverage/coverage/proc/self/cwd/source/common/upstream/outlier_detection_impl.cc.html

Reply:
Huh, I think this _should_ be covered by the tests I added in this PR already - this function is expected to be called with both of these ejection types on lines 1124 and 1235. I'll do some digging into why the coverage build isn't seeing this path get exercised.

Reply:
Yeah, something's definitely fishy with the coverage build here. There are a _ton_ of `EXPECT_CALL(*event_logger_, logEject(...))` statements in the outlier detection tests, but the coverage build only reports `logEject` being called twice. It appears to be called twice directly in `OutlierDetectionEventLoggerImplTest` on lines 1551 and 1591, so I'd guess that's where the coverage report is seeing the calls. Am I misunderstanding how the coverage report works, maybe?

Reply:
I've not seen it being incorrect before, but I've seen plenty of cases where something is mocked out so my code isn't called.  You can always sanity check by putting in an ASSERT(0) temporarily and see if the tests fail or not :-)

Reply:
Aha, that does look like it. I'll add a bit to `OutlierDetectionEventLoggerImplTest` to test the new path.